### PR TITLE
[RNMobile] Support Theme Colors and Gradients

### PR DESCRIPTION
**Pull request by chipsnyder**
**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/1744

**Related PRs**
`gutenberg-mobile` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2241
`WordPress-Android` https://github.com/wordpress-mobile/WordPress-Android/pull/12041
`WordPress-FluxC-Android` https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1593
`WordPress-iOS` https://github.com/wordpress-mobile/WordPress-iOS/pull/14085

----

## Description

Adds support for theme defined colors and gradients to be injected into the editor from the mobile apps. 

Colors can then be displayed by the same mechanisms that utilize the settings lookup for colors and gradients. This is currently supported on Cover and Button blocks

## How has this been tested?

***Note*** Atomic sites don't seem to support the `wp/v2/themes` API changes yet so this can be tested with a free site or a self-hosted site

To help with testing this theme can be added to a self-hosted site: [twentytwenty-copy.zip](https://github.com/wordpress-mobile/WordPress-iOS/files/4657974/twentytwenty-copy.zip)
### Colors
1.) Select a theme with custom colors (such as TwentyTwenty)
2.) Create a post with blocks that have a custom color set (such as Cover or Button)
3.) Load editor on mobile 

**Expect** to see the custom color on the block

### Gradients
1.) Select a theme with custom gradients or [add gradients to a theme](https://github.com/WordPress/gutenberg/blob/12a6ee49c16dd6f327a8a141b6858197536da800/docs/designers-developers/developers/themes/theme-support.md#block-gradient-presets)
2.) Create a post with blocks that have a custom gradient set (such as Cover or Button)
3.) Load editor on mobile 

**Expect** to see the custom gradient on the block

### No Wifi
1.) Set up blocks for custom color and gradient
2.) View the post on mobile to cache the theme
3.) Turn off wifi to the device
4.) Reload the editor

**Expect** to see the custom colors on the blocks

### No Wifi - Force kill the app
1.) Set up blocks for custom color and gradient
2.) View the post on mobile to cache the theme
3.) Turn off wifi to the device
4.) Stop the app from running in the background
5.) Reload the editor

**Expect** to see the custom colors on the blocks

## Screenshots 
<kbd></kbd>

## Types of changes

New feature: Theme support

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by chipsnyder** [comment]
@dratwas same thing here as the question [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2241#issuecomment-625974387) whenever you have a minute

**Comment by dratwas** [comment]
@chipsnyder I'm afraid that it will not work with the color picker since we use the defaults from `block-editor` here - https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/block-settings/container.native.js#L14

I think that the colors from settings should be used in the color-picker too. You should be able to test that since this feature is merged already. Just open button settings and check if the palette is correct.

**Comment by chipsnyder** [comment]
Thanks for the guidance @dratwas. I should be able to refactor that to pull from the updated settings instead of from the default pre-loaded ones. I'll work on that adjustment now.

**Comment by pinarol** [comment]
Is it possible to use the primary color of theme when we first create the button? Currently we are still using gray.

It can be worth exploring web side to find out how they decide which one is the primary color.

**Comment by chipsnyder** [comment]
> It can be worth exploring web side to find out how they decide which one is the primary color.

This would probably be the main challenge but I'll do some digging and see what I can find.

**Comment by chipsnyder** [comment]
Looking through the code and at the twenty-twenty theme, for buttons, it seems to default to what's defined in the theme's class `wp-block-button__link.`

Twenty-Twenty theme:
```
.editor-styles-wrapper .wp-block-button__link,
.editor-styles-wrapper .wp-block-file__button {
	background: #cd2653;
	border-radius: 0;
	color: #fff;
	font-size: 15px;
	font-weight: 600;
	letter-spacing: 0.0333em;
	line-height: 1.25;
	padding: 1.1em 1.44em;
	text-transform: uppercase;
}
```
So for this case it will display `#cd2653` which is the Accent Color. This also happens to be the first color in the list of colors but that's just a coincidence. 

We could default to displaying the first color set in `editor-color-palette` without any trouble. But this would be very theme dependent that those settings are coordinated.

**Comment by lukewalczak** [comment]
I agree with @chipsnyder, `Button` should display `accent` color, which should be also the first color in palette.

**Comment by pinarol** [comment]
The case where Button is created on mobile is easier to resolve as far as I see, we can initially set the first color in the palette to the block attributes that way it'll be displayed correctly on web as well.

Let's decide what to do if the Button is created on web without any color info. What should mobile display? @iamthomasbishop was mentioning we could show a warning message somewhere in this case. Can you describe where that message could be @iamthomasbishop ?

In addition to the warning message which color do you think we should fallback to? @iamthomasbishop

1. Gray color we currently have(it'll make it more explicit that the color can not be resolved)
2. 1st color of the palette since it looks like for most of the cases it matches the web, it is just not guaranteed. (but we'll have a warning message? 🤷‍♀️ )

**Comment by pinarol** [comment]
@youknowriad We can really use your opinion on this. Do you think it makes sense to create Button blocks with an initial palette color on web? Currently they have no color info that mobile can render until somebody sets a color for them.

**Comment by youknowriad** [comment]
> We can really use your opinion on this. Do you think it makes sense to create Button blocks with an initial palette color on web?

I'm not certain. Themes might have a default color and a palette to change it. It's all theme specific. I guess you can't load the default CSS on mobile which makes this hard. Maybe we can accept that mobile apps are not wysiwyg and just use the default gray buttons. It's not ideal but I suspect that this issue will come up for any block and not just for the background color which means trying to solve it in adhoc way won't get us that far

**Comment by pinarol** [comment]
Thanks for the context @youknowriad .

**Comment by pinarol** [comment]
>  I guess you can't load the default CSS on mobile which makes this hard.

@youknowriad one more question on this, what do you mean by `default CSS` here?

**Comment by youknowriad** [comment]
I mean the theme's CSS that style button for instance. (editor styles)

**Comment by chipsnyder** [comment]
I opened a new issue for the button enhancement (https://github.com/wordpress-mobile/gutenberg-mobile/issues/2278) so that we can come up with our path forward and continue to discuss that in a different thread.

**Comment by chipsnyder** [comment]
No new updates were needed here for the Android side so I went ahead and updated this with master, @dratwas can you take a look for me?

File: packages/edit-post/src/editor.native.js [hunk 1]
```diff
@@ -70,6 +72,14 @@ class Editor extends Component {
...
 		}
 
+		if ( colors !== undefined ) {
+			settings.colors = colors;
+		}
+
+		if ( gradients !== undefined ) {
+			settings.gradients = gradients;
+		}
+
```
[the remark this answers is not in the document]
Comment (dratwas) [reply packages/edit-post/src/editor.native.js:82]: Yeah! Good catch!

**Review by dratwas — APPROVED** [review]
Good job @chipsnyder 🚀

**Pull request merged**